### PR TITLE
Hide cookie banner for WP apps

### LINKED ISSUE #31158: Disable cookie warnings when viewing a site within the mobile apps
#### Steps to reproduce
1. Use the mobile app and navigate to "My Sites"
2. Select a site and scroll down to "View My Site"
3. It should show you your site

#### What I expected
My website without the the cookie warning at the bottom

#### What happened instead
The cookie warning is shown for a site I own

#### Browser / OS version
WordPress iOS and Android app

#### Screenshot / Video

img_7489

#### Context / Source

There is no need to show a cookie warning for a site we own within the app.

#dogfooding

### PR DESCRIPTION
**Pull request by obenland**
#### Changes proposed in this Pull Request
* Hide the cookie banner on requests coming from the mobile app (we're already in the app, so we don't need to show this banner).

Screen Shot 2019-03-28 at 3 41 31 PM

#### Testing instructions

* Open the reader in the calypso.live environment of this PR (you won't have dismissed the cookie banner there)
* Use a fake user agent that contains either `wp-iphone` or `wp-android`:
  * iOS example: `Mozilla/5.0 (iPhone; CPU iPhone OS 12_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/16B91 wp-iphone/12.1`.
  * Android example: `Mozilla/5.0 (Linux; Android 6.0; Android SDK built for x86_64 Build/MASTER; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/44.0.2403.119 Mobile Safari/537.36 wp-android/4.7`.
  * On Chrome you can do this using the Network conditions on the browser devtools:
Screen Shot 2019-03-28 at 13 01 49
Screen Shot 2019-03-28 at 13 02 45

* Reload the page.
* Make sure the cookie banner doesn't show up now.

Fixes #31158.

### CODE REVIEW

File: client/blocks/gdpr-banner/index.jsx [hunk 1]
```diff
@@ -53,6 +54,9 @@ class GdprBanner extends Component {
 		if ( cookies.sensitive_pixel_option === 'yes' || cookies.sensitive_pixel_option === 'no' ) {
 			return false;
 		}
+		if ( isMobileApp() ) {
+			return false;
+		}
```
Comment (dmsnell) [inline client/blocks/gdpr-banner/index.jsx]: Is it right to hide the GDPR banner on mobile web views? do we have a native mobile view that presents the warning?

**Comment by mmtr** [comment]
I rebased this branch after merging #31830, so this PR contains now only the needed changes :)

**Comment by mmtr** [comment]
This works great @obenland !

I however have concerns about what implications can cause this change in terms of GPDR, so I asked for feedback on the original issue: https://github.com/Automattic/wp-calypso/issues/31158#issuecomment-477828985.

**Comment by rclations** [comment]
Just noting for reference that the consent banner in the screenshot above is a completely separate consent banner (different code, different purpose) than the screenshot shown in #31158. Not sure if that was intentional or not.

**Comment by obenland** [comment]
Thanks, the banner from #31158 was removed in r189618-wpcom

**Pull request merged**